### ISSUE TITLE
Major regression!! - Nightly OF/OF release not generating project files

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/projectGenerator#565]
I am guessing this is a PG issue but the current OF nightly release isn't making project files.

<img width="625" alt="image" src="https://github.com/user-attachments/assets/054aa1da-5fa5-43bb-8817-eb8a6e386aad">

Not sure if this is connected to #555 or #554 

In the nightly log I see: 

```
Compiling projectGenerator for Debug
make[1]: Entering directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
💿 Using CCACHE -- config.project.mk 
make[2]: warning: -j2 forced in submake: resetting jobserver mode.
make[2]: Entering directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
make[2]: Leaving directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
make[2]: warning: -j2 forced in submake: resetting jobserver mode.
make[2]: Entering directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
cp ../../../libs/*/lib/linux64/*.so bin/ ; true

     compiling done
     to launch the application

     cd bin
     ./projectGenerator_debug
     
     - or -
     
     make RunDebug


make[2]: Leaving directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
make[1]: Leaving directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
Creating project files for osx
projectGenerator does not exist. Continue.
```

which corresponds to these lines: 

```
        if [ -f "${main_ofroot}/apps/projectGenerator/commandLine/bin/projectGenerator" ]; then
            echo "projectGenerator exists..."
            ${main_ofroot}/apps/projectGenerator/commandLine/bin/projectGenerator --recursive -p${pg_platform} -o$pkg_ofroot $pkg_ofroot/examples > /dev/null
        else
            echo "projectGenerator does not exist. Continue."
        fi
```

https://github.com/openframeworks/openFrameworks/blob/master/scripts/dev/create_package.sh#L196-L201

Maybe the pg is not building as `projectGenerator` but `commandLine`? 

cc @danoli3 @dimitre

**Comment by ofTheo** [comment]
Okay I listed what's in the commandLine.bin/  folder and its due the the debug being added to the end of the name. 

```
make[1]: Leaving directory '/home/runner/work/openFrameworks/openFrameworks/apps/projectGenerator/commandLine'
Creating project files for linux64
projectGenerator does not exist. Continue.
libfmod.so
projectGenerator_debug
```

I think as a quick fix I will change it to use the release build so it will generate `projectGenerator` instead of `projectGenerator_debug` 

Note: the regression probably originated in this submodule, but I think it's fine to correct the create_package script and it should speed up the nightly build package generation step.

**Comment by dimitre** [comment]
Maybe it is because of ```make Debug``` commands here
but I'm not famliar with the script

https://github.com/openframeworks/openFrameworks/blob/37fd61b2ccf5d3bf0fdac1f01c11bf0ab0f3dd68/scripts/dev/create_package.sh#L178-L185

**Comment by dimitre** [comment]
we should change <json.hpp> to <nlohmann/json.hpp> inside pg
related:
https://github.com/openframeworks/openFrameworks/issues/8082

**Comment by dimitre** [comment]
https://github.com/openframeworks/projectGenerator/pull/566

**Comment by ofTheo** [comment]
Thanks @dimitre - #566 probably doesn't fix the error I was seeing right? 
Just something that needed updating?

**Comment by dimitre** [comment]
Right. I didnt' read with attention and thought it was the case.
I was puzzled by the error and about the ```what() ``` explanation, so I'm trying to find exceptions that only echoes e.what() and detail them more, so we know where the error is.

**Comment by dimitre** [comment]
ok solved finally haha

**Comment by dimitre** [comment]
it seems now to be generating, but for some reason they are not readable by xcode. 
from what I'm seeing the only change is an extra pair of [ ] enclosing the entire json
nlohmann json seems to have changed recently

**Comment by danoli3** [comment]
xcode project reading needs no comments in the file, if any change to the xcode project that adds any comments or symbols that are not json it will fail

**Comment by danoli3** [comment]
there are no changes to json headers since December 2023. where release is 
![image](https://github.com/user-attachments/assets/6cd32b55-a4e9-4716-a81c-cdb75343796a)

**Comment by danoli3** [comment]
exit failure on linux when trying to backup .vscode

**Comment by danoli3** [comment]
working on fixes.
looks like errors with addons and all sorts of cmdLine pg stuff

**Comment by ofTheo** [comment]
Just to confirm what @dimitre mentioned. 
I just ran a manual nightly build and the project files are there now, but Xcode `project.pbxproj` contents is wrapped with the extra [   ]. 

Removing that it all works fine.

**Comment by dimitre** [comment]
yes or use something along the lines ```json[0].dump```

**Comment by dimitre** [comment]
I had to change PG to detect '[' and usse dump accordingly.
it seems to fix nightly I've just downloaded and it is opening ok.
the issue is, nlohmann::json is behaving differently in different systems or versions(I think) so we have to consider both possible outputs.

**Comment by dimitre** [comment]
maybe related: https://github.com/nlohmann/json/issues/4320

**Comment by ofTheo** [comment]
@dimitre that issue definitely seems related! good find!! 
I'll close this issue for now - thanks for the fix!!

**Comment by ofTheo** [comment]
closed by #568
